src/actions/accounts.js

```diff
@@ -16,14 +16,35 @@
 
 'use strict'
 
+import { Alert, Keyboard } from 'react-native'
+import { Actions } from 'react-native-router-flux'
 import {
-  ADD_ACCOUNT, SELECT_ACCOUNT, DELETE_ACCOUNT, MODIFY_ACCOUNT, SET_NEW_PIN, SET_ACCOUNTS
+  ADD_ACCOUNT, SELECT_ACCOUNT, DELETE_ACCOUNT, MODIFY_ACCOUNT, SET_NEW_PIN, SET_OLD_PIN, SET_ACCOUNTS
 } from '../constants/AccountActions'
+import { encryptData, decryptData } from '../util/native'
 
-export function addAccount (account) {
-  return {
-    type: ADD_ACCOUNT,
-    account
+export function addAccount (pin) {
+  return async function (dispatch, getState) {
+    try {
+      let account = getState().accounts.selected
+      if (!account) {
+        return
+      }
+
+      let seed = await encryptData(account.seed, pin)
+      dispatch({
+        type: ADD_ACCOUNT,
+        account: {
+          encryptedSeed: seed,
```

Comment:
ADD_ACCOUNT is not overwriting `selected` account so it won't contain `encryptedSeed` property required for instance to call `setOldPin` action.